src/plugins/data/public/search/aggs/types.ts

```diff
@@ -70,3 +72,25 @@ export interface SearchAggsStart {
   ) => InstanceType<typeof AggConfigs>;
   types: AggTypesRegistryStart;
 }
+
+/** @internal */
+export interface AggExpressionType {
+  type: 'agg_type';
+  value: CreateAggConfigParams;
+}
+
+/** @internal */
+export type AggExpressionFunctionArgs<
+  Name extends keyof AggParamsMapping
+> = AggParamsMapping[Name] & Pick<AggConfigJson, 'id' | 'enabled' | 'schema'>;
+
+/**
+ * A global list of the param interfaces for each agg type.
+ * For now this is internal, but eventually we will probably
+ * want to make it public.
+ *
+ * @internal
+ */
+export interface AggParamsMapping {
+  terms: AggParamsTerms;
+}
```

Comment:
We can add each new agg type here, along with the params for that agg. This way you can access all the agg params from this mapping.

Eventually we can make this interface public, and folks could even add to it using `declare module` to append their agg to the interface. (This would be necessary if, for example, there were a special type of aggregation which was only provided by x-pack).